### DISCUSSION TITLE
Redis Streams

### DISCUSSION THREAD

**Discussion by t3r-cni** [hibiken/asynq#418] [General]
Note: I am very new to this library.  
I am curious if there is any future plans to include streams into getting and receiving tasks from a queue. 

It is possible it doesn't make sense with the current architecture. Am I correct in understanding that the way workers get and receive tasks is through a rest request? I wonder if there is a faster way to do this, like a stream.

**Comment by hibiken** [comment]
@t3r-cni Thank you for opening a discussion! (And sorry for the delay in responding 🙏 )

Currently, I don't have plans to using Redis stream since current implementation of using Redis list, sorted-sets, etc works quite well for our use case. 

> Am I correct in understanding that the way workers get and receive tasks is through a rest request? I wonder if there is a faster way to do this, like a stream.

Current implementation of worker [polls redis list for pending tasks](https://github.com/hibiken/asynq/blob/04d7c8c38cf9919d33eccbc0495a583d31451488/processor.go#L174,191). Ideally we'd like to use `BRPOPLPUSH` command here to avoid polling but this command currently doesn't support watching multiple list (https://github.com/redis/redis/issues/1785)

**Reply by t3r-cni** [reply]
Thanks for the response!
